mlflow/types/utils.py

```diff
@@ -78,6 +87,185 @@ def _get_str_or_byte_type(data):
         return DataType.binary
 
 
+def _infer_datatype(data) -> Optional[DataType]:
+    if DataType.is_boolean(data):
+        return DataType.boolean
+    if DataType.is_integer(data):
+        return DataType.integer
+    if DataType.is_long(data):
+        return DataType.long
+    if DataType.is_float(data):
+        return DataType.float
+    if DataType.is_double(data):
+        return DataType.double
+    if DataType.is_string(data):
+        return DataType.string
+    if DataType.is_binary(data):
+        return DataType.binary
+    if DataType.is_datetime(data):
+        return DataType.datetime
+
+
+def _merge_object_properties(obj1: Object, obj2: Object) -> Object:
```

Comment:
Can we implement this in `Object` as `merge`?

Reply:
Why do we want to do that? I think this should be an internal method that's not exposed to users.

Reply:
Yeah, it should be an internal method like `_merge`.

Reply:
But why do we want to put it into `Object` instead of just in utils.py as a utility helper function.

Reply:
To me, it looks more natural to define how `Object` instances should be merged in `Object`.

Reply:
Updated!